src/middleware/types.ts

```diff
@@ -138,42 +139,23 @@ export interface InteractiveMessage<Action extends InteractiveAction> extends Ke
   original_message?: { [key: string]: string; };
 }
 
-export interface Container {
-  message_ts: string;
-  channel_id: string;
-  is_ephemeral: boolean;
-  thread_ts?: string;
-}
-
-export interface MessageContainer extends Container {
-  type: 'message';
-}
-
-export interface MessageAttachmentContainer extends Container {
-  type: 'message_attachment';
-  attachment_id: number;
-  is_app_unfurl: boolean;
-  app_unfurl_url?: string;
-}
-
-export type KnownActions = UsersSelectResponse | StaticSelectResponse | ConversationsSelectResponse |
-ChannelsSelectResponse | ExternalSelectResponse | ButtonResponse | OverflowResponse | DatepickerResponse;
-
-export interface ActionResponse {
+// this interface is "abstract", and not meant to be used on its own. it just helps DRY up the following interfaces
+interface BaseElementAction {
   block_id: string;
   action_id: string;
   action_ts: string;
+  // TODO: figure out when this is actually available
```

Comment:
Confirm is available for any action element. As more action elements are added, this may change.

Reply:
im starting to think the idea of having a `BaseElementAction` type isn't a great idea. its useful for reducing repetition, but really, the type system should be figuring out the "common" parts for us. we should just be able to define the various event payloads, and the common parts are just in the union by evaluation. i think that would be more clear, and less prone to error of certain properties flowing into other interfaces.